connectors/src/connectors/slack/bot.ts

```diff
@@ -176,32 +179,37 @@ async function botAnswerMessage(
   // We start by retrieving the slack user info.
   const slackClient = await getSlackClient(connector.id);
 
-  const userOrBotId = slackUserId || slackBotId;
-  if (!userOrBotId) {
-    throw new Error("Failed to get slack user id or bot id");
+  let slackUserInfo: SlackUserInfo | null = null;
+  if (slackBotId) {
+    slackUserInfo = await getSlackBotInfo(slackClient, slackBotId);
+  } else if (slackUserId) {
+    slackUserInfo = await getSlackUserInfo(slackClient, slackUserId);
   }
-  const slackUserInfo: SlackUserInfo = await getSlackUserOrBotInfo(
-    slackClient,
-    userOrBotId
-  );
 
   if (!slackUserInfo) {
     throw new Error("Failed to get slack user info");
   }
 
-  const hasChatbotAccess = await notifyIfSlackUserIsNotAllowed(
-    connector,
-    slackClient,
-    slackUserInfo,
-    {
-      slackChannelId: slackChannel,
-      slackTeamId,
-      slackMessageTs,
-    },
-    slackConfig.whitelistedDomains
-  );
-  if (!hasChatbotAccess) {
-    return new Ok(undefined);
+  if (slackUserInfo.is_bot) {
```

Comment:
Any reason to extract this logic from the main function `notifyIfSlackUserIsNotAllowed` that handles all the permissions logic?

Reply:
I think it make sense to separate bot permissions and user permissions.
Also, the code path for user permissions does not follow the existing error model:
We return  a `new Err(SlackExternalUserError('message..'))` everywhere to notify the user, but the function that checks the permissions uses `slack.chat.postMessage()` directly and return undefined. We should fix that in a separate PR.

Reply:
Fair... My point is more that having some of the permission living in `botAnswerMessage` while some other logic lives in `notifyIfSlackUserIsNotAllowed` is not ideal.. While `notifyIfSlackUserIsNotAllowed` might not be the best name for it, the important thing is that the logic is centralized. The more the logic is spread out, the higher the risk of overlooking something.

Reply:
Ok, lets deploy that and I will create a task to fix that in a follow up PR.
We have an important customer waiting on it.

Reply:
(task created on my personal todo on Slack, will get back to it 💯)